components/builder-jobsrv/src/server/log.rs

```diff
@@ -0,0 +1,48 @@
+use error::Result;
+use std::fs;
+use std::path::PathBuf;
+
+/// Encapsulates the local filesystem directory in which in-process
+/// build job logs will be collected prior to being sent to long-term
+/// storage.
+#[derive(Clone, Debug)]
+pub struct LogDirectory {
+    path: PathBuf
+}
+
+impl LogDirectory {
+
+    /// Create a new `LogDirectory` wrapping `path`.
+    pub fn new(path: &str) -> Self {
+        LogDirectory { path: PathBuf::from(path) }
+    }
+
+    /// Ensures that the specified log directory can be used.
+    ///
+    /// # Panics
+    /// * when the path does not exist
+    /// * when the path is not a directory
+    /// * when the path is not writable
+    pub fn validate(&self) -> Result<()> {
+        let meta = fs::metadata(&self.path)
+            .expect(format!("Build log directory {:?} doesn't exist!",
```

Comment:
Is there a specific reason to not just create the dir if it doesn't exist?  That is the pattern we seem to use more in the codebase.